assets/src/application/ui/layout/entityList/filterBar/filterState/useEntityFilterService.ts

```diff
@@ -15,55 +15,63 @@ const useEntityFilterService = <D extends string, L extends string, E extends En
 
 	const { getFilters, getSearches, getSorters } = useFilterBarService<D, L, E, ELFSM>(domain, listId);
 
-	const getCallbackList = (
-		mappedCallbackList: ReturnType<typeof getFilters>
-	): Array<SubscriptionCallback<FilterBarServiceCbArgs<E, ELFSM>, E[]>> => {
+	const getCallbackList = useCallback((mappedCallbackList: ReturnType<typeof getFilters>): Array<
+		SubscriptionCallback<FilterBarServiceCbArgs<E, ELFSM>, E[]>
+	> => {
 		const subscriptions = sortBy(pathOr(10, ['options', 'priority']), Object.values(mappedCallbackList));
 		return subscriptions.map(({ callback }) => callback);
-	};
+	}, []);
 
-	const applyCallbacks = (
-		entityList: Array<E>,
-		filterState: FS,
-		mappedCallbackList: ReturnType<typeof getFilters>
-	): Array<E> => {
-		let filteredEntities = entityList;
+	const applyCallbacks = useCallback(
+		(entityList: Array<E>, filterState: FS, mappedCallbackList: ReturnType<typeof getFilters>): Array<E> => {
+			let filteredEntities = entityList;
 
-		const callbacks = getCallbackList(mappedCallbackList);
+			const callbacks = getCallbackList(mappedCallbackList);
 
-		callbacks.forEach((callback) => {
-			filteredEntities = callback({ entityList: filteredEntities, filterState });
-		});
+			callbacks.forEach((callback) => {
+				filteredEntities = callback({ entityList: filteredEntities, filterState });
+			});
 
-		return filteredEntities;
-	};
+			return filteredEntities;
+		},
+		[getCallbackList]
+	);
 
-	const applyFilters: EFS['applyFilters'] = useCallback(
+	// avoid the callback being affected by change in other callbacks
+	const filterIdsStr = Object.keys(getFilters()).join(':');
+	const applyFilters = useCallback<EFS['applyFilters']>(
 		(entityList, filterState) => {
 			return applyCallbacks(entityList, filterState, getFilters());
 		},
-		[domain, listId]
+		[applyCallbacks, filterIdsStr]
```

Comment:
what the?!?! those previous values were not even used in the callback !?!?!

is there any chance that `getFilters` will change?
I can't see those changing dynamically...

Reply:
Yes, it will. For example if your register a sorter, it changes the subscriptions hash thus affection all (filters, sorters and searches). What we do here is that we update the dep to only change when a new filter is registered or updated.